src/reducers/example.tsx

```diff
@@ -0,0 +1,54 @@
+import { Reducer } from 'redux';
+
+const TOGGLE_ON = 'TOGGLE_ON';
+const TOGGLE_OFF = 'TOGGLE_OFF';
+
+export interface ExampleState {
+  toggle: 'on' | 'off';
+}
+
+interface ToggleOnAction {
```

Comment:
Have you looked at [typesafe-actions](https://github.com/piotrwitek/typesafe-actions)? I've seen it mentioned in a few of the examples I've looked at for Redux+TS and it looks pretty good. I think it could cut down on verbosity of defining actions.

Reply:
I had not looked at this. I like it! Let's use it. It cut the code size down by nearly 50% and I think it still reads well. The API is well documented, too.

```
2 files changed, 12 insertions(+), 21 deletions(-)
```